src/Compilers/CSharp/Portable/Operations/CSharpCompoundAssignmentOperation.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal abstract class BaseCSharpCompoundAssignmentOperation : BaseCompoundAssignmentExpression
+    {
+        private static readonly CommonConversion CSharpConversion = new CommonConversion(exists: true, isIdentity: true, isNumeric: false, isReference: false, methodSymbol: null);
+
+        protected BaseCSharpCompoundAssignmentOperation(Operations.BinaryOperatorKind operatorKind, bool isLifted, bool isChecked, IMethodSymbol operatorMethod, SemanticModel semanticModel, SyntaxNode syntax, ITypeSymbol type, Optional<object> constantValue, bool isImplicit) : base(operatorKind, isLifted, isChecked, operatorMethod, semanticModel, syntax, type, constantValue, isImplicit)
+        {
+        }
+
+        public override CommonConversion InConversion => CSharpConversion;
+        public override CommonConversion OutConversion => CSharpConversion;
```

Comment:
>public override CommonConversion OutConversion => CSharpConversion; [](start = 8, length = 67)

This doesn't look right, C# can have non-identity conversions in both places.
For example:
```
class Program
{
    static void Main(string[] args)
    {
        var x = new C();
        var y = new C();

        x += y;
    }
}

class C
{
    public static int operator +(int x, C y)
    {
        return 0;
    }

    public static implicit operator int (C x)
    {
        return 0;
    }

    public static implicit operator C(int x)
    {
        return null;
    }
}
```